src/caret_analyze/architecture/architecture.py

```diff
@@ -501,13 +509,15 @@ def remove_variable_passing(self, node_name: str,
         callback_write: CallbackStructValue = \
             Util.find_one(lambda x: x.callback_name == callback_name_write, self.callbacks)
 
-        if callback_read.publish_topic_names:
+        if callback_read.publish_topics:
             context_reader = AssignContextReader(node)
-            for publish_topic_name in callback_read.publish_topic_names:
-                if callback_write.subscribe_topic_name:
+            for publish_topic in callback_read.publish_topics:
+                if callback_write.subscribe_topic_name and publish_topic is not None:
                     context_reader.remove_callback_chain(callback_write.subscribe_topic_name,
-                                                         publish_topic_name)
-
+                                                         callback_write.
+                                                         subscription_construction_order,
+                                                         publish_topic.topic_name,
+                                                         publish_topic.construction_order)
```

Comment:
Suggestion to improve readability by removing line break after period

```suggestion
                    context_reader.remove_callback_chain(
                        callback_write.subscribe_topic_name,
                        callback_write.subscription_construction_order,
                        publish_topic.topic_name,
                        publish_topic.construction_order
                    )
```